jcl/src/java.base/share/classes/com/ibm/oti/util/Util.java

```diff
@@ -290,19 +290,34 @@ public static void appendLnTo(Appendable buf) {
  */
 public static void printStackTraceElement(StackTraceElement e, Object elementSource, Appendable buf, boolean extended) {
 	/*[IF Sidecar19-SE]*/
-	final java.lang.String modName = e.getModuleName();
-	if (null != modName) {
+	boolean classLoaderIncluded = false;
+	final String classLoaderName = e.getClassLoaderName();
+	if ((null != classLoaderName) && (!classLoaderName.isEmpty())) {
+		final ClassLoader classLoader = e.getClassLoader();
+		// Only include the classloader name in the trace if it isn't a built-in classloader
+		if (!isBuiltInClassLoader(classLoader)) {
+			classLoaderIncluded = true;
+			appendTo(buf, classLoaderName);
+			appendTo(buf, "/"); //$NON-NLS-1$
+		}
+	}
+
+	final String modName = e.getModuleName();
+	if ((null == modName) || (modName.isEmpty())) {
+		// If the classloader name was included but there is no module name, note the empty module in the trace
+		if (classLoaderIncluded) {
+			appendTo(buf, "/"); //$NON-NLS-1$
+		}
+	} else {
 		appendTo(buf, modName);
 		if (extended) {
```

Comment:
Could you print the module version even with the extended = false?
This way I guess #11447 could be fixed too. Or is it better to open another PR?

Reply:
@Thihup I haven't reproduced the issue in #11447 on my end yet. I'll follow up in that issue. Should be able to add the fix in here if it works out :)